OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```diff
@@ -86,7 +86,10 @@ public void Attack(Actor targetActor, bool allowMove = true)
 			if (!targetActor.HasTrait<FrozenUnderFog>() && !Self.Owner.CanTargetActor(targetActor))
 				Log.Write("lua", "{1} is not revealed for player {0}!", Self.Owner, targetActor);
 
-			attackBase.AttackTarget(target, true, allowMove);
+			if (attackBase == null)
+				Log.Write("lua", "{0} does not have an Attack* trait. Ignoring order to attack {1}.".F(Self, targetActor));
```

Comment:
Our convention is to throw a script error if the map tries to do something bogus.